py/runtime.c

```diff
@@ -1023,6 +1023,8 @@ STATIC mp_obj_t mp_obj_new_checked_fun(const mp_obj_type_t *type, mp_obj_t fun)
 // see http://docs.python.org/3/howto/descriptor.html
 // and also https://mail.python.org/pipermail/python-dev/2015-March/138950.html
 void mp_convert_member_lookup(mp_obj_t self, const mp_obj_type_t *type, mp_obj_t member, mp_obj_t *dest) {
+    // clear dest[1] to indicate no special attribute found yet
+    dest[1] = MP_OBJ_NULL;
```

Comment:
Why exactly is this line needed?  What are the cases/scenarios where things would fail if it were removed?  Could this line be moved below, eg to just before the call to `type->attr(...)`?

Reply:
Hmm, I think I understand a bit better what this is doing.  Can it be moved to below the check for SENTINEL, like:
```
      // dest[1] == MP_OBJ_SENTINEL means continue lookup, otherwise return
        if (dest[1] != MP_OBJ_SENTINEL) {
            return;
        }
        dest[1] = MP_OBJ_NULL;
```
?

Reply:
Good idea. I've rebased and updated the PR accordingly.